File: apps/cli/src/commands/auth/status.ts
/**
 * Status command for the LTF CLI
 * Shows current authentication status
 */

import { Command } from 'commander';
import { getAuthStatus } from '../../lib/auth.js';
import { getConfigPath, getContext } from '../../lib/config.js';
import output from '../../lib/output.js';

/**
 * Format expiry time for display
 */
function formatExpiry(expiresAt: Date | undefined): string {
  if (!expiresAt) {
    return output.colors.muted('Never');
  }

  const now = new Date();
  const diff = expiresAt.getTime() - now.getTime();

  if (diff < 0) {
    return output.colors.error('Expired');
  }

  const hours = Math.floor(diff / (1000 * 60 * 60));
  const minutes = Math.floor((diff % (1000 * 60 * 60)) / (1000 * 60));

  if (hours > 24) {
    const days = Math.floor(hours / 24);
    return `${days} day${days > 1 ? 's' : ''} (${expiresAt.toLocaleString()})`;
  }

  if (hours > 0) {
    return `${hours}h ${minutes}m (${expiresAt.toLocaleString()})`;
  }

  return `${minutes} minute${minutes !== 1 ? 's' : ''} (${expiresAt.toLocaleString()})`;
}

/**
 * Execute the status command
 */
export async function statusAction(): Promise<void> {
  const status = getAuthStatus();
  const context = getContext();

  output.header('Authentication Status');

  if (status.authenticated) {
    output.keyValue([
      ['Status', output.colors.success('Authenticated')],
      ['Email', status.email || output.colors.muted('Not available')],
      ['User ID', status.userId || output.colors.muted('Not available')],
      ['Token type', status.type === 'clerk' ? 'Clerk OAuth' : 'API token'],
      ['Expires', formatExpiry(status.expiresAt)],
    ]);

    // Show warning if token is expiring soon (within 1 hour)
    if (status.expiresAt) {
      const timeLeft = status.expiresAt.getTime() - Date.now();
      if (timeLeft < 60 * 60 * 1000 && timeLeft > 0) {
        output.newline();
        output.warning('Your session will expire soon. Run `ltf auth login` to refresh.');
      }
    }

    // Show current project context if set
    if (context?.projectId) {
      output.newline();
      output.header('Active Context');
      output.keyValue([
        ['Workspace', context.workspaceName || context.workspaceId || output.colors.muted('None')],
        ['Project', context.projectName
          ? `${context.projectName} (${context.projectKey})`
          : context.projectId || output.colors.muted('None')],
      ]);
    }
  } else {
    output.keyValue([
      ['Status', output.colors.error('Not authenticated')],
    ]);
    output.newline();
    output.info('Run `ltf auth login` to authenticate');
  }

  // Show config file location
  output.newline();
  output.log(output.colors.muted(`Config file: ${getConfigPath()}`));
}

/**
 * Create and return the status command
 */
export function createStatusCommand() {
  const statusCommand = new Command('status')
    .description('Show current authentication status')
    .action(statusAction);

  return statusCommand;
}

export default createStatusCommand;

File: apps/tui/internal/commands/auth/status.go
package auth

import (
	"fmt"
	"time"

	"github.com/spf13/cobra"

	"github.com/aansh-afk/ltf1-pm/apps/tui/internal/api"
	"github.com/aansh-afk/ltf1-pm/apps/tui/internal/output"
)

func newStatusCmd() *cobra.Command {
	return &cobra.Command{
		Use:   "status",
		Short: "Show authentication status",
		RunE: func(cmd *cobra.Command, args []string) error {
			cfg, err := api.LoadAuthConfig()
			if err != nil || cfg == nil || cfg.Auth.Token == "" {
				output.Warningf("not authenticated. Run: ltf1 auth login")
				return nil
			}

			output.Header("Authentication")
			fmt.Printf("Email:        %s\n", cfg.Auth.Email)
			fmt.Printf("User ID:      %s\n", cfg.Auth.UserID)
			fmt.Printf("Token type:   %s\n", cfg.Auth.TokenType)
			if cfg.Auth.ExpiresAt > 0 {
				expiry := time.UnixMilli(int64(cfg.Auth.ExpiresAt))
				if api.IsAuthenticated(cfg) {
					fmt.Printf("Expires:      %s\n", expiry.Format("2006-01-02 15:04"))
				} else {
					fmt.Printf("Expired at:   %s\n", expiry.Format("2006-01-02 15:04"))
				}
			}

			if api.HasProjectContext(cfg) {
				output.Header("Active Project")
				fmt.Printf("Workspace:    %s\n", cfg.Context.WorkspaceName)
				fmt.Printf("Project:      %s (%s)\n", cfg.Context.ProjectName, cfg.Context.ProjectKey)
			} else {
				output.Warningf("no project selected. Run: ltf1 project select")
			}
			return nil
		},
	}
}
